shunit2

```diff
@@ -1285,11 +1285,24 @@ oneTimeSetUp
 command [ $? -eq ${SHUNIT_TRUE} ] \
     || _shunit_fatal "oneTimeSetUp() returned non-zero return code."
 
-# Execute the suite function defined in the parent test script.
-# DEPRECATED as of 2.1.0.
-suite
+# Command line selected tests or suite selected tests
+if command [ "$#" -ge 2 ]; then
+  # Argument $1 is either the filename of tests or '--'; either way, skip it.
+  shift
+  # Remaining arguments ($2 .. $#) are assumed to be test function names.
+  # Interate through all remaining args in "$@" in a POSIX (likely portable) way.
+  # Helpful tip: https://unix.stackexchange.com/questions/314032/how-to-use-arguments-like-1-2-in-a-for-loop
+  for _shunit_arg_ do
+    suite_addTest "${_shunit_arg_}"
+  done
+  unset _shunit_arg_
+else
+  # Execute the suite function defined in the parent test script.
+  # DEPRECATED as of 2.1.0.
+  suite
+fi
 
-# If no suite function was defined, dynamically build a list of functions.
+# If no tests specified defined, dynamically build a list of functions.
 if command [ -z "${__shunit_suite}" ]; then
```

Comment:
Not sure I follow what you are saying.  As it stands (with my changes), you can run a test script either:

1. without arguments, which implicitly runs all things named `test*`, or the test suite if one exists, or
2. with arguments, e.g. `test_script.sh test_foo test_bar`, in which case only those (two) tests will run.

Are you saying if both `test_foo` and `test_bar` do not exist, NO tests will run?  That's sort of the case, as it will actually try to run those functions and fail as they don't exist.  My changes do no checking to see if the test functions specified exist.  

Is there a third option I'm not seeing?

Reply:
That sounds good. I misunderstood the code on this line and was reacting to the comment

> If no tests specified defined,

Is "defined" there a typo?

Reply:
Yeah, that's a residual typo - fixed in 584f14636e6f08cb1430499486657fa5ba06404d.

Also added documentation to the README.md that I should have written the first time around.